src/core/mac/mac.cpp

```diff
@@ -1265,6 +1265,12 @@ void Mac::HandleTransmitDone(TxFrame &aFrame, RxFrame *aAckFrame, otError aError
         otDumpDebgMac("TX", aFrame.GetHeader(), aFrame.GetLength());
         FinishOperation();
         Get<MeshForwarder>().HandleSentFrame(aFrame, aError);
+#if OPENTHREAD_THREAD_VERSION >= OPENTHREAD_THREAD_VERSION_1_2
+        if (aError == OT_ERROR_NONE && aAckFrame != NULL)
+        {
+            Get<DataPollSender>().ResetKeepAliveTimer();
```

Comment:
As we are only using EKA for secured frames, we should confirm security was used here before resetting the timer.

Reply:
In 1.1, data poll is used as keep alive, the ack is not secured. I think this is also acceptable in 1.2. Thoughts?

Reply:
I believe @CiaranWoodward's point here is we should check if the sent frame has security enabled before resetting the keep alive timer. I agree with that comment.

Reply:
Then shall we make this clear in the spec, I remember it states **any** MAC Data Frame can be used as keep alive frame.

Reply:
Yes, this comment was raised a few weeks ago in this PR, [SPEC-908](https://threadgroup.atlassian.net/browse/SPEC-908) was submitted as a result, and you acknowledged it in https://github.com/openthread/openthread/pull/4325#discussion_r348293011.

Reply:
Oh, I see. Thank you!